tests/handover.cpp

```diff
@@ -0,0 +1,370 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include "catch.hpp"
+
+#include "realm/commit_log.hpp"
+#include "util/test_file.hpp"
+
+#include "list.hpp"
+#include "object_accessor.hpp"
+#include "object_schema.hpp"
+#include "property.hpp"
+#include "schema.hpp"
+#include "handover.hpp"
+
+#include <thread>
+#include <future>
+
+using namespace realm;
+
+static TableRef get_table(Realm& realm, const ObjectSchema &object_schema) {
+    StringData name (("class_" + object_schema.name).c_str());
```

Comment:
read_group() is public and is there specifically because everything that uses the object store needs to be able to call the ObjectStore functions on a Realm's group.